pkg/abaputils/abaputils.go

```diff
@@ -100,18 +101,27 @@ func ReadServiceKeyAbapEnvironment(options AbapEnvironmentOptions, c command.Exe
 
 	if err != nil {
 		// Executing cfReadServiceKeyScript failed
-		return abapServiceKey, err
+		return abapServiceKeyV8.Credentials, err
+	}
+
+	// Depending on the cf cli version, the service key may be returned in a different format. For compatibility reason, both formats are supported
+	json.Unmarshal([]byte(serviceKeyJSON), &abapServiceKeyV8)
+	if abapServiceKeyV8 == (AbapServiceKeyV8{}) {
+		log.Entry().Debug("Could not parse the service key in the cf cli v8 format.")
+	} else {
+		log.Entry().Info("Service Key read successfully")
+		return abapServiceKeyV8.Credentials, nil
 	}
 
-	// parse
 	json.Unmarshal([]byte(serviceKeyJSON), &abapServiceKey)
 	if abapServiceKey == (AbapServiceKey{}) {
-		log.SetErrorCategory(log.ErrorInfrastructure)
-		return abapServiceKey, errors.New("Parsing the service key failed. Service key is empty")
+		log.Entry().Debug("Could not parse the service key in the cf cli v7 format.")
+	} else {
+		log.Entry().Info("Service Key read successfully")
+		return abapServiceKey, nil
 	}
-
-	log.Entry().Info("Service Key read successfully")
-	return abapServiceKey, nil
+	log.SetErrorCategory(log.ErrorInfrastructure)
+	return abapServiceKeyV8.Credentials, errors.New("Parsing the service key failed for all supported formats. Service key is empty")
```

Comment:
Maybe the error from json.Unmarshal of the v8 attempt could be provided as well as it might contain hints if future format changes occur

Reply:
The error is mostly nil. But added a debug log line, if the error returns something useful.